daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -840,7 +841,7 @@ private[lf] object SBuiltin {
     override private[speedy] final def execute(
         args: util.ArrayList[SValue],
         machine: Machine,
-    ): Unit = {
+    ): Unit = machine.withOnLedger("SBUCreate") { onLedger =>
```

Comment:
Why not defining a an abstract class for `OnLedgerBuiltin` ?  Something like :+1: 
```scala
sealed abstract class OnLedgerBuiltin(arity: Int) extends SBuiltin(arity) with Product {
    
    protected def execute(
        args: util.ArrayList[SValue],
        machine: Machine,
        onLedger: OnLedger
    ): Unit

    final override def execute(args: util.ArrayList[SValue], machine: Machine): Unit =
      machine.withOnLedger(productPrefix)(execute(args, machine, _))
  }
```

Reply:
sounds good, let me give it a try

Reply:
Added.

Reply:
:+1: